phabricator.py

```diff
@@ -37,8 +39,25 @@ def run(self):
         # Find the preselected branch
         git_branch = settings.get('branch')
 
-        # If no preselected branch is provided
-        if git_branch == None:
+        if git_branch is None and settings.get('use_arc_land_onto_default', False):
+            # Get current branch
+            arc_args = [settings.get('arc_path', 'arc'), 'get-config', 'arc.land.onto.default']
+            arc_child = subprocess.Popen(
+                arc_args, cwd=filedir,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            # DEV: We decode for Python 3 which receives bytes
+            arc_stdout = arc_child.stdout.read().decode('utf-8')
+            arc_stderr = arc_child.stderr.read().decode('utf-8')
+            if arc_stderr:
+                print('Ran `{0}` in `{1}`'.format(' '.join(arc_args), filedir))
+                print('STDERR: {0}'.format(arc_stderr))
+
+            # Grep the output to find the return value
+            m = re.search('.*Current Value: "(?P<value>.*)"\n.*', arc_stdout)
```

Comment:
Does `arc` have anything like git's porcelain output for `get-config`? It would be nice to avoid a gnarly regexp like this as it will likely break due to internationalization.